Module10.playground/Contents.swift

```diff
@@ -0,0 +1,62 @@
+import Foundation
+import PlaygroundSupport
+
+PlaygroundPage.current.needsIndefiniteExecution = true
+
+struct CatFacts: Decodable {
+  let fact: String
+  let length: Int
+}
+
+struct CatFactsAsyncSequence: AsyncSequence {
+  typealias Element = CatFacts
+  
+  struct AsyncIterator: AsyncIteratorProtocol {
+    private var remainingFacts: Int
+    
+    init(count: Int) {
+      self.remainingFacts = count
+    }
+    
+    mutating func next() async throws -> CatFacts? {
+      guard remainingFacts > 0 else { return nil }
+      
+      let url = URL(string: "https://catfact.ninja/fact")!
```

Comment:
This can be a private property. There's no need to build it every iteration since it doesn't change.